### PR TITLE
Docs: Add GitLab CI/CD to integrations.

### PR DESCRIPTION
**Pull request by jvacek**
## Summary
Adds a sample cicd config for gitlab

### CODE REVIEW

**Comment by zanieb** [comment]
Related https://docs.astral.sh/uv/guides/integration/gitlab/

File: docs/integrations.md [hunk 1]
```diff
@@ -156,3 +156,25 @@ In addition, ruff publishes the following images:
...
+  interruptible: true
+  image:
+    name: ghcr.io/astral-sh/ruff:0.7.1-alpine
+  variables:
+    CODE_LOCATION: $CI_PROJECT_DIR
+  script:
+    - cd $CODE_LOCATION
+    - ruff --version
+    - ruff check --output-format=gitlab > code-quality-report.json
+    - ruff format --diff --verbose
```
Comment (zanieb) [inline docs/integrations.md:176]: Why the verbose flag on `format`?

Comment (jvacek) [reply docs/integrations.md:176]: Ah yeah that's probably not necessary indeed, I do like to see which configs are being taken into account if I'm puzzled by a fail
```suggestion
    - ruff format --diff
```

**Comment by jvacek** [comment]
Not sure if it's preferrable to split this up into two separate jobs so they can run in parallel? Guess the users can figure that out for themselves in my opinion.

File: docs/integrations.md [hunk 2]
```diff
@@ -156,3 +156,25 @@ In addition, ruff publishes the following images:
 
 As with the distroless image, each image is published with ruff version tags as
 `ruff:{major}.{minor}.{patch}-{base}` and `ruff:{major}.{minor}-{base}`, e.g., `ruff:0.6.6-alpine`.
+
+## GitLab CI/CD
```
Comment (MichaReiser) [inline docs/integrations.md:160]: Should we move this right after GitHub Actions to have all CI in one place?

File: docs/integrations.md [hunk 3]
```diff
@@ -156,3 +156,25 @@ In addition, ruff publishes the following images:
...
+
+You can add the following configuration to `.gitlab-ci.yml` to run a `ruff format`, and a `ruff check` compatible with GitLab's codequality.
+
+```yaml
+Ruff:
+  stage: .pre
+  interruptible: true
+  image:
+    name: ghcr.io/astral-sh/ruff:0.7.1-alpine
```
Comment (MichaReiser) [inline docs/integrations.md:169]: What's the motivation of using the docker image over using e.g. pip or uv pip to install the project's ruff version? 
That would be more aligned with our GitHub Actions example

File: docs/integrations.md [hunk 4]
```diff
@@ -156,3 +156,25 @@ In addition, ruff publishes the following images:
...
 As with the distroless image, each image is published with ruff version tags as
 `ruff:{major}.{minor}.{patch}-{base}` and `ruff:{major}.{minor}-{base}`, e.g., `ruff:0.6.6-alpine`.
+
+## GitLab CI/CD
+
+You can add the following configuration to `.gitlab-ci.yml` to run a `ruff format`, and a `ruff check` compatible with GitLab's codequality.
+
+```yaml
+Ruff:
+  stage: .pre
```
Comment (MichaReiser) [inline docs/integrations.md:166]: Should we use the `build` stage to ensure the job runs even if it is the only job?

> If a pipeline contains only jobs in the .pre or .post stages, it does not run. There must be at least one other job in a different stage. ([source](https://docs.gitlab.com/ee/ci/yaml/index.html#stage-pre))

**Review by MichaReiser — COMMENTED** [review]
I haven't used GitLab in ages. So I'm sorry for any obvious questions.

File: docs/integrations.md (hunk 4 above)
Comment (jvacek) [reply docs/integrations.md:166]: Yeah fair, not a great idea to assume _anything_ about the end-users setups hah

File: docs/integrations.md (hunk 3 above)
Comment (jvacek) [reply docs/integrations.md:169]: It's _way_ faster than installing ruff with `pip` every single time. That's one of the major advantages of the new images with extra tags you guys push.

File: docs/integrations.md (hunk 2 above)
Comment (jvacek) [reply docs/integrations.md:160]: If you prefer to, sure

**Review by jvacek — COMMENTED** [review]
> Not sure if it's preferrable to split this up into two separate jobs so they can run in parallel? Guess the users can figure that out for themselves in my opinion. 

I split this up, let me know if you'd prefer the other

**Review by MichaReiser — APPROVED** [review]
This looks good to me. 

Thanks a lot for contributing the workflow template!

File: docs/integrations.md (hunk 3 above)
Comment (MichaReiser) [reply docs/integrations.md]: That makes sense. It does have the downside that they have to update the ruff version number locally in the project dependencies and CI. But I guess we can always add a second example if needed.

**Pull request merged**